espflash/src/connection.rs

```diff
@@ -119,43 +125,77 @@ impl Connection {
         Ok(reset_after_flash(&mut self.serial, pid)?)
     }
 
-    // Reset the device to flash mode
-    pub fn reset_to_flash(&mut self, extra_delay: bool) -> Result<(), Error> {
-        if self.port_info.pid == USB_SERIAL_JTAG_PID {
-            self.serial.write_data_terminal_ready(false)?;
-            self.serial.write_request_to_send(false)?;
+    fn classic_reset(&mut self, extra_delay: bool) -> Result<(), Error> {
+        info!(
+            "Attempting Classic reset with {} delay...",
+            if extra_delay { "extra" } else { "default" }
+        );
 
-            sleep(Duration::from_millis(100));
+        self.serial.write_data_terminal_ready(false)?;
+        self.serial.write_request_to_send(false)?;
 
-            self.serial.write_data_terminal_ready(true)?;
-            self.serial.write_request_to_send(false)?;
+        self.serial.write_data_terminal_ready(true)?;
+        self.serial.write_request_to_send(true)?;
```

Comment:
Where does this come from?  [esptool doesn't do this.](https://github.com/espressif/esptool/blob/master/esptool/reset.py#L61) (?)

Also, since in `unix_tight_reset` we have the `IO0/EN` comments, it would be good to have them here too

Reply:
This is the only difference I can discern between this branch and the branch I have trying to solve this issue. If these 4 lines are all it takes to get my branch working that would be great, but I would like to be certain this does not have detrimental effects on other OSs.

Reply:
This bit of code essentially turns classic reset into unix tight reset. This seems to work because of the way the retrying is working currently:

https://github.com/esp-rs/espflash/pull/387/files#diff-9b8216d416248667f4612fb114829cb89c5754ca818d5784b3b0d879ec0d4ee6R70-R75

Here, we should be retrying the UNIX reset for X attempts, and then moving on to the classic mode.